java/core/src/main/java/com/microsoft/bond/Bonded.java

```diff
@@ -135,4 +134,14 @@
      * @return a new Bonded instance constrained to the given type or null if this conversion is not possible
      */
     public abstract <U extends BondSerializable> Bonded<U> convert(StructBondType<U> toBondType);
+
+    /**
+     * Equality on Bonded and Bonded subtypes is always reference equality
+     * (i.e., the same as Object.equals()). The semantics of .equals() in
+     * codegen'd types depend on this.
+     */
+    @Override
+    public final boolean equals(Object other) {
+        return this == other;
+    }
```

Comment:
Please seal hashCode as well. Also, you can just call super, whose equals/hashCode do exactly what we need.